x-pack/platform/packages/shared/kbn-streamlang/src/conditions/condition_to_painless.ts

```diff
@@ -38,10 +38,114 @@ function encodeValue(value: StringOrNumberOrBoolean | null | undefined) {
   return value;
 }
 
+// Helper function to evaluate date math expressions in Painless
+function evaluateDateMath(expression: string): string {
+  const expr = String(expression).trim();
+
+  // Check if it's a date math expression (contains 'now' or '||')
+  if (expr.includes('now') || expr.includes('||')) {
+    let code: string;
+    let remainingExpr: string;
+
+    // Parse the anchor date and any subsequent operations
+    if (expr.startsWith('now')) {
+      // Anchor is 'now'
+      code = 'System.currentTimeMillis()';
+      remainingExpr = expr.substring(3); // Remove 'now'
+    } else if (expr.includes('||')) {
+      // Anchor is a date string with ||
+      const [dateStr, mathExpr] = expr.split('||');
+
+      // Normalize various date formats to ISO format for Painless
+      const normalizedDate = normalizeDateString(dateStr.trim());
+
+      // Parse the date string to get milliseconds
+      code = `Instant.parse(${encodeValue(normalizedDate)}).toEpochMilli()`;
+      remainingExpr = mathExpr || '';
+    } else {
+      // Not a valid date math expression
+      const encoded = encodeValue(String(expression));
+      return typeof encoded === 'string' ? encoded : String(encoded);
+    }
+
+    // Milliseconds per unit for offset calculations
+    const msPerUnit: Record<string, number> = {
+      y: 365.25 * 24 * 60 * 60 * 1000, // approximate year
```

Comment:
The comments suggest this was already thought about, but there is potential for this to be slightly off vs Elasticsearch's date math.

ES will calculate a month as 1 calendar month (28-31 days depending on the month), this PR will add 30 days. Similar story for years, ES will use 365 or 366 (leap years), but this calculation would work out slightly different.